pkg/server/grpc_server_test.go

```diff
@@ -1182,6 +1188,94 @@ func TestAPIWithIssuerClaimConfig(t *testing.T) {
 	}
 }
 
+// Tests API with an RSA key
```

Comment:
I didn't want to unit test the algorithm registry code too heavily here since that's `sigstore/sigstore`'s responsibility.

I'm focusing on:
- Testing that other key types work (RSA in this case).
- Testing that non-permitted algorithms are rejected (ECDSA with P521 curve in this case).
- Testing both the CSR and non-CSR paths.